File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.cpp
bool comp(int a,int b)
{
    return a>b;
}
class Solution {
public:
    int minSetSize(vector<int>& arr) {
        vector<int>mp(100005);
        for(int i = 0;i<arr.size();i++)
            mp[arr[i]]++;
        sort(mp.begin(),mp.end(),comp);
        int cnt = 0,n = arr.size(),num=0;
        for(int i = 0;i<100001;i++){
            cnt+=mp[i];
            num++;
            if(cnt>=(n/2))
                return num;
        }
        return num;
        
    }
};

File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.py
class Solution(object):
    def minSetSize(self, arr):
        l = [0]*100001
        for i in arr:
            l[i]+=1
        l.sort(reverse = True)
        n = len(arr)
        cnt = 0
        nos = 0
        for i in l:
            cnt = cnt + i
            nos = nos + 1
            if cnt >= (n/2):
                return nos
        return 0;
        